core/src/main/java/org/apache/calcite/rel/type/RelDataTypeFactory.java

```diff
@@ -191,6 +193,24 @@ RelDataType createTypeWithNullability(
       RelDataType type,
       boolean nullable);
 
+  /**
+   * Creates a type that is the same as another type but with possibly
+   * different nullability. The output type may be identical to the input
+   * type. For type systems without a concept of nullability, the return value
+   * is always the same as the input.  This differs from {@link
+   * #createTypeWithNullability(RelDataType, boolean)} in the handling of struct
+   * types.  This function returns a nullable struct.
+   *
+   * @param type     input type
```

Comment:
Nit: I see that some javadoc here has this kind of formatting with extra spaces to align columns "horizontally", AFAICT this is not what we generally use in the codebase, so I'd rather not enforce this here